### PR TITLE
feat: support trigger command to trigger a recommed manually

### PR DESCRIPTION
**Pull request by mangoGoForward**
## What type of PR is this?
Feature

We can manually trigger a recommend like:

```shell
kubectl-crane recommend trigger --name workloads-ntzns -n default
```

### CODE REVIEW

File: pkg/cmd/recommend/trigger.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+		return fmt.Errorf("failed to get related recommendation rule, %v", err)
+	}
+
+	// Obtain the runNumber from the recommendationRule
+	runNumber := recommendationRule.Status.RunNumber
+
+	// Modify the run number annotation for the recommendation that should be triggered.
+	// The run number should be lower than the runNumber in the recommendationRule.
+	runNumber -= 1
+	recommend.Annotations[RunNumberAnnotation] = strconv.Itoa(int(runNumber))
```
Comment (qmhu) [inline pkg/cmd/recommend/trigger.go]: Please make sure recommend.Annotations != nil here.

File: pkg/cmd/recommend/trigger.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	if err != nil {
+		return fmt.Errorf("failed to get related recommendation rule, %v", err)
+	}
+
+	// Obtain the runNumber from the recommendationRule
+	runNumber := recommendationRule.Status.RunNumber
+
+	// Modify the run number annotation for the recommendation that should be triggered.
+	// The run number should be lower than the runNumber in the recommendationRule.
+	runNumber -= 1
```
Comment (qmhu) [inline pkg/cmd/recommend/trigger.go]: Better to set runNumber to 0.

File: pkg/cmd/recommend/trigger.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+func (o *RecommendTriggerOptions) Run() error {
+	recommend, err := o.CommonOptions.CraneClient.AnalysisV1alpha1().Recommendations(*o.CommonOptions.ConfigFlags.Namespace).Get(context.TODO(), o.Name, metav1.GetOptions{})
+	if err != nil {
+		return errors.New("the recommend doesn't exist, please specify a existed recommend name with --name")
+	}
+
+	// get related recommendation rule
+	recommendationRuleName, ok := recommend.Labels[RecommendationRuleNameLabel]
```
Comment (qmhu) [inline pkg/cmd/recommend/trigger.go]: No need to get recommendationRule's runNumber ,just set annotation to 0 and exist.

File: pkg/cmd/recommend/trigger.go (hunk 1 above)
Comment (mangoGoForward) [reply pkg/cmd/recommend/trigger.go]: Done

File: pkg/cmd/recommend/trigger.go (hunk 2 above)
Comment (mangoGoForward) [reply pkg/cmd/recommend/trigger.go]: Done

File: pkg/cmd/recommend/trigger.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+)
+
+var (
+	triggerRecommendExample = `
+# manually trigger the specified recommendation rule
+%[1]s recommend trigger --name workloads-ntzns -n default
+
+# pre-commit
+%[1]s recommend trigger --name workloads-ntzns -n default --dry-run
+`
```
Comment (qmhu) [inline pkg/cmd/recommend/trigger.go:23]: 1. "manually trigger the specified recommendation rule " change to "manually trigger the specified recommendation "
2. "workloads-ntzns" looks like a workload name , we need a recommendation name here

Comment (mangoGoForward) [reply pkg/cmd/recommend/trigger.go:23]: Done

**Review by qmhu — APPROVED** [review]
LGTM

**Pull request merged**